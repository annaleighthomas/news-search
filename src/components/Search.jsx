import React from 'react';
import PropTypes from 'prop-types';


const Search = ({ search, onChange, onSubmit }) => {

  return (<>
    <form onSubmit={onSubmit}>
      <input name="search" value={search} onChange={onChange} />
      <button>Submit</button>
    </form>
  </>);

};

Search.propTypes = {
  search: PropTypes.string,
  onChange: PropTypes.func,
  onSubmit: PropTypes.func,
};

export default Search;
