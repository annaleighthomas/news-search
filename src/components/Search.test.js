import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Search from './Search';

describe('Search component', () => {
  afterEach(() => cleanup());
  it('renders Search component', () => {
    const { asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
  });
});
