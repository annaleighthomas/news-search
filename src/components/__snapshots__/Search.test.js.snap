// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders Search component 1`] = `
<DocumentFragment>
  <form>
    <input
      name="search"
      value=""
    />
    <button>
      Submit
    </button>
  </form>
</DocumentFragment>
`;
